src/datomic_toolbox/core.clj

```diff
@@ -229,3 +229,24 @@
         {:created-at (first tx-instants)
          :updated-at (last tx-instants)
          :timestamps tx-instants})))
+
+(defn retry-tx
```

Comment:
But what is a transaction but transaction data given to `db/transact`?

Couldn't this take the transaction data instead of `f`, replacing `(f)` with `@(db/transact tx-data)`? That would make the macro unnecessary.

Taking an arbitrary function doesn't, to me, make it seem like it's retrying a transaction. It's retrying, well, anything.